Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+
+        self.setPropertySettings('SampleGeometryFile', EnabledWhenProperty('SampleGeometry', PropertyCriterion.IsEqualTo, 'Custom'))
+
+        self.declareProperty(name="DetectorEfficiencyCalibration",
+                             defaultValue="None",
+                             validator=StringListValidator(["None", "Vanadium", "Incoherent",  "Paramagnetic"]),
+                             direction=Direction.Input,
+                             doc="Detector efficiency calibration type.")
+
+        self.setPropertySettings('DetectorEfficiencyCalibration', sample)
+
+        self.declareProperty(name="IncoherentCrossSection",
+                             defaultValue=0,
+                             validator=IntBoundedValidator(lower=0),
+                             direction=Direction.Input,
+                             doc="Cross-section for incoherent scattering, necessary for setting the output on the absolute scale.")
+
+        incoherent = EnabledWhenProperty('DetectorEfficiencyCalibration', PropertyCriterion.IsEqualTo, 'Incoherent')
+
+        absoluteNormalisation = EnabledWhenProperty('AbsoluteUnitsNormalisation', PropertyCriterion.IsDefault)
+
+        self.setPropertySettings('IncoherentCrossSection', EnabledWhenProperty(incoherent, absoluteNormalisation,
+                                                                               LogicOperator.And))
+
+        self.declareProperty(name="OutputUnits",
+                             defaultValue="Wavelength/TOF",
+                             validator=StringListValidator(["Wavelength/TOF", "TwoTheta", "Q"]),
+                             direction=Direction.Input,
+                             doc="The choice to display the reduced data either as a function of the raw data units, the detector twoTheta,"
+                                 +" or the momentum exchange.")
+
+        self.setPropertySettings('OutputUnits', EnabledWhenProperty(vanadium, sample, LogicOperator.Or))
+
+        self.declareProperty(name="TOFUnits",
+                             defaultValue="TimeChannels",
+                             validator=StringListValidator(["TimeChannels", "UncalibratedTime", "Energy"]),
+                             direction=Direction.Input,
+                             doc="The choice to display the TOF data either as a function of the time channel or the uncalibrated time.\
+                             It has no effect if the measurement mode is monochromatic.")
+
+        tofMeasurement = EnabledWhenProperty('MeasurementTechnique', PropertyCriterion.IsEqualTo, 'TOF')
+
+        self.setPropertySettings('TOFUnits', EnabledWhenProperty(tofMeasurement,
+                                                                 EnabledWhenProperty(vanadium, sample, LogicOperator.Or),
+                                                                 LogicOperator.And))
+
+        self.declareProperty(FileProperty('InstrumentParameterFile', '',
```

Comment:
Is this the YIG? If so make it clear.

Reply:
This is the YIG calibration output, as the doc should suggest. I've changed the name to a more explicit: 'InstrumentCalibration'.